File: app/src/main/java/com/rsupport/mobile1/test/activity/ImageRepository.java
package com.rsupport.mobile1.test.activity;

import androidx.lifecycle.LiveData;
import java.util.ArrayList;

// 이미지 URL을 가져오는 레포지토리 클래스
public class ImageRepository {
    private RemoteDataSource remoteDataSource;

    // 생성자
    public ImageRepository() {
        remoteDataSource = new RemoteDataSource();
    }

    // 현재 페이지에 해당하는 이미지 URL을 가져오는 메서드
    public LiveData<ArrayList<String>> getImageUrls(int currentPage) {
        return remoteDataSource.getImageUrls(currentPage);
    }
}

File: app/src/main/java/com/rsupport/mobile1/test/activity/data/ImageRepository.kt
package com.rsupport.mobile1.test.activity.data

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import org.jsoup.Jsoup
import javax.inject.Inject

interface ImageRepository {
    fun getImageUrlList(page: Int): Flow<List<String>>
}


class ImageRepositoryImpl @Inject constructor(): ImageRepository {

    override fun getImageUrlList(page: Int): Flow<List<String>> = flow {
        val url = "https://www.gettyimages.com/photos/collaboration?assettype=image&sort=mostpopular&phrase=collaboration&license=rf,rm&page=$page"
        val doc = Jsoup.connect(url).get()
        val imageUrls = doc
            .select("picture")
            .select("img[src]")
            .map { element -> element.attr("src") }
        emit(imageUrls)
    }.flowOn(Dispatchers.IO)
}